stages/org.osbuild.dracut

```diff
@@ -0,0 +1,159 @@
+
+NB: needs chroot for now as well as `strip` for stripping the initrfams.
+"""
+STAGE_OPTS = """
+"required": ["kernel"],
+"properties": {
+  "kernel": {
+    "description": "List of target kernel versions",
+    "type": "array",
+    "items": {
+       "type": "string",
+       "description": "A kernel version"
+    }
+  },
+  "modules": {
+    "description": "Exact list of dracut modules to use.",
+    "type": "array",
+    "items": {
+       "type": "string",
+       "description": "A dracut module, e.g. base, nfs, network ..."
+    }
+  },
+  "add_modules": {
+    "description": "Additional dracut modules to include.",
+    "type": "array",
+    "items": {
+      "type": "string",
+      "description": "A dracut module, e.g. base, nfs, network ..."
+    }
+  },
+  "omit_modules": {
+    "description": "Dracut modules to not include.",
+    "type": "array",
+    "items": {
+      "type": "string",
+      "description": "A dracut module, e.g. base, nfs, network ..."
+    }
+  },
+  "drivers": {
+    "description": "Kernel modules to exclusively include.",
+    "type": "array",
+    "items": {
+      "type": "string",
+      "description": "A kernel module without the .ko extension"
+    }
+  },
+  "add_drivers": {
+    "description": "Add a specific kernel modules.",
+    "type": "array",
+    "items": {
+      "type": "string",
+      "description": "A kernel module without the .ko extension"
+    }
+  },
+  "force_drivers": {
+    "description": "Add driver and ensure that they are tried to be loaded.",
+    "type": "array",
+    "items": {
+      "type": "string",
+      "description": "A kernel module without the .ko extension"
+    }
+  },
+  "filesystems": {
+    "description": "Kernel filesystem modules to exclusively include.",
+    "type": "array",
+    "items": {
+      "type": "string",
+      "description": "A kernel module without the .ko extension"
+    }
+  },
+  "include": {
+    "description": "Add custom files to the initramfs.",
+    "type": "array",
+    "items": {
+      "type": "object",
+      "description": "What (keys) to include where (values)"
+    }
+  }
+}
+"""
+
+
+def yesno(name: str, value: bool) -> str:
+    prefix = "" if value else "no-"
+    return f"--{prefix}{name}"
+
+
+def main(tree, options):
+    kernels = options["kernel"]
+    modules = options.get("modules", [])  # dracut modules
+    add_modules = options.get("add_modules", [])
+    omit_modules = options.get("omit_modules", [])
+    drivers = options.get("drivers", [])  # kernel modules
+    add_drivers = options.get("add_drivers", [])
+    force_drivers = options.get("force_drivers", [])
+    filesystems = options.get("filesystems", [])
+    include = options.get("include", [])
+    early_microcode = options.get("early_microcode", False)
+
+    # internal option for exploring purposes
+    host_only = options.get("host-only", False)
+
+    # initrds have already been created, force the recreation
```

Comment:
Do we still want this after your patch that allows initrd generation to be disabled in the rpm stage?

Reply:
Is there any harm? Avoid accidental re-creation of the initramfs?